start_vscode_bessemer.sh

```diff
@@ -0,0 +1,479 @@
+                shift
+                shift
+                ;;
+                *)
+                echo -e "Warning: ignoring unknown option $1 \n"
+                shift
+                ;;
+        esac
+done
+
+###############################################################################
+# Check configuration options                                                 #
+###############################################################################
+
+# check if user has a configuration file and source it to initialize options
+if [ -f "$VSC_CONFIG_FILE" ]; then
+        echo -e "Found configuration file $VSC_CONFIG_FILE"
+        echo -e "Initializing configuration from file ${VSC_CONFIG_FILE}:"
+        cat "$VSC_CONFIG_FILE"
+        source "$VSC_CONFIG_FILE"
+fi
+
+# check that VSC_USERNAME is not an empty string
+if [ -z "$VSC_USERNAME" ]
+then
+        echo -e "Error: No TUoS username is specified, terminating script\n"
+        display_help
+else
+        echo -e "TUoS username: $VSC_USERNAME"
+fi
+
+# check number of CPU per task
+
+# check if VSC_CPUS_PER_TASK
+if ! [[ "$VSC_CPUS_PER_TASK" =~ ^[0-9c]+$ ]]; then 
+        echo -e "Error: $VSC_CPUS_PER_TASK -> Incorrect format. Please specify number of tasks per node as an integer and try again\n"
+        display_help
+fi
+
+# check if VSC_CPUS_PER_TASK is <= 40
+if [ "$VSC_CPUS_PER_TASK" -gt "40" ]; then
+        echo -e "Error: $VSC_CPUS_PER_TASK -> Larger than 40. No distributed memory supported, therefore the number of cpus per task needs to be smaller or equal to 40\n"
+        display_help
+fi
+
+if [ "$VSC_CPUS_PER_TASK" -gt "0" ]; then
+        echo -e "Requesting $VSC_CPUS_PER_TASK cpus per task for running the code-server"
+        display_help
+fi
+
+# check number of GPUs
+
+# check if VSC_NUM_GPU an integer
+if ! [[ "$VSC_NUM_GPU" =~ ^[0-9]+$ ]]; then
+        echo -e "Error: $VSC_NUM_GPU -> Incorrect format. Please specify the number of GPU as an integer and try again\n"
+        display_help
+fi
+
+# check if VSC_NUM_GPU is <= 4
+if [ "$VSC_NUM_GPU" -gt "4" ]; then
+        echo -e "Error: No distributed memory supported, therefore number of GPUs needs to be smaller or equal to 4\n"
+        display_help
+fi
+
+if [ "$VSC_NUM_GPU" -gt "0" ]; then
+        echo -e "Requesting $VSC_NUM_GPU GPUs for running the code-server"
+        VSC_SNUM_GPU="--gpus-per-node=$VSC_NUM_GPU"
+else
+        VSC_SNUM_GPU=""
+fi
+# check if VSC_PARTITION_ID is set
+if [ "$VSC_NUM_GPU" -gt "0" ] && {[ "$VSC_PARTITION_ID" == "gpu"  ] || [ "$VSC_PARTITION_ID" == "gpu-a100-tmp"  ]} ; then
+        echo -e "Requesting partition $VSC_PARTITION_ID"
+        VSC_SNUM_GPU="--partition=$VSC_PARTITION_ID --qos=gpu $VSC_SNUM_GPU"
+else
+        if [ "$VSC_NUM_GPU" -gt "0" ]; then
+                echo -e "Error: partition incorrect. Please specify either gpu or gpu-a100-tmp"
+                display_help
+        fi      
+fi
+
+if [ "$VSC_CPUS_PER_TASK" -gt "0" -a ! "$VSC_NUM_GPU" -gt "0" ]; then
```

Comment:
This evaluates wrong due to a missing !

```suggestion
if [ ! "$VSC_CPUS_PER_TASK"-gt "0" -a ! "$VSC_NUM_GPU" -gt "0" ]; then
```

Reply:
Ah yes thanks. I had spotted this. Fixed. Cheers